src/main/java/edu/kis/powp/jobs2d/drivers/macro/MacroDriverObserver.java

```diff
@@ -0,0 +1,23 @@
+package edu.kis.powp.jobs2d.drivers.macro;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DriverFeature;
+import edu.kis.powp.jobs2d.features.MacroFeature;
+import edu.kis.powp.observer.Subscriber;
+
+import java.util.List;
+
+public class MacroDriverObserver implements Subscriber {
+
+    @Override
+    public void update() {
+        Job2dDriver currentDriver = DriverFeature.getDriverManager().getCurrentDriver();
+        if (!(currentDriver instanceof MacroDriverDecorator) && MacroFeature.getMacroDriverDecorator().isFlagActive()) {
+            MacroDriverDecorator driver = MacroFeature.getMacroDriverDecorator();
+            driver.setDriver();
```

Comment:
This part of the code is hard to decode. Moreover, MacroDriverDecorator seems to be unnecessarily tangled with DriverManager and DriverFeature (see: DriverFeature.getDriverManager() call in setDriver implementation). 
Please clean up those parts of the code.